pkg/provisioner/provisioner.go

```diff
@@ -84,8 +84,9 @@ type Provisioner interface {
 	PowerOn() (result Result, err error)
 
 	// PowerOff ensures the server is powered off independently of any image
-	// provisioning operation.
-	PowerOff() (result Result, err error)
+	// provisioning operation. The boolean argument may be used to specify
+	// if a hard reboot (force power off) is required - true if so.
+	PowerOff(hardMode bool) (result Result, err error)
```

Comment:
Instead of a bool, this argument should be the same type used to define the `Mode` field. That way if we add more modes, we don't have to change the argument.

Reply:
Likewise, `hasRebootAnnotation()` would want to return a mode instead of a "hard" bool

Reply:
Thanks @beekhof and @dhellmann - I've updated this now.